src/main/java/hu/oe/nik/szfmv/automatedcar/visualization/CourseDisplay.java

```diff
@@ -112,19 +115,35 @@ private AutomatedCar getCarObject(List<WorldObject> objects) {
      * @param g2d Buffered image.
      * @param world World object which will be drawn.
      */
-    private void drawObjects(Graphics2D g2d, World world) {
-        int[] offsets = getCarOffset(getCarObject(world.getWorldObjects()));
+    private void drawObjects(Graphics2D g2d, WorldManager world) {
+        int[] offsets = getCarOffset(world.getAutomatedCar());
 
         DebugViewer viewer = new DebugViewer(g2d);
-
-        for (WorldObject object : world.getWorldObjects()) {
+        AutomatedCar car = world.getAutomatedCar();
+        for (IObject object : world.getAllObjectsInRectangle(
+                new Position(0,0),
+                new Position(this.world.getWidth(),this.world.getHeight()))) {
+            Point2D refPoint;
+            try {
+                refPoint = new Point(object.getReferenceX(), object.getReferenceY());
+            } catch (NullPointerException e){
+                refPoint = new Point(0,0);
+            }
             AffineTransform t = new AffineTransform();
-            t.translate(object.getX() + offsets[0], object.getY() + offsets[1]);
+            t.translate(object.getPosX() - refPoint.getX() + offsets[0], object.getPosY() - refPoint.getY()+ offsets[1]);
+            t.rotate(Math.toRadians(-object.getRotation()),refPoint.getX(),refPoint.getY());
             g2d.drawImage(object.getImage(), t, this);
 
             // todo: decide on how model will signal colors
             viewer.DrawPolygon(object.getX()+ offsets[0], object.getY() + offsets[1], object.getWidth(), object.getHeight());
         }
+
+        // Draw car
+
+        AffineTransform t1 = new AffineTransform();
+        t1.translate(car.getPosX()+ offsets[0], car.getPosY()+ offsets[1]);
+        g2d.drawImage(car.getImage(), t1, this);
+        t1.rotate(Math.toRadians(car.getRotation()),offsets[0],offsets[1]);
```

Comment:
since we get the objects around the car, the car will always be in the list which we draw, so there is no need to draw it again

Reply:
Not necessarily, it’s going to be stored separately in the world manager